### ISSUE TITLE
QueryArgs missed the caller

### ISSUE THREAD

**Issue by DamianStraszak** [Cardinal-Cryptography/ink-wrapper#81]
It seems to me that https://github.com/Cardinal-Cryptography/ink-wrapper/blob/d13339dd7ed1f591effe863b99532e239e84c2a7/ink-wrapper-types/src/calls.rs#L231 should have one additional parameter: the contract caller. This way when dry-running one can pass any account.

**Comment by obrok** [comment]
I agree with the goal of dry-running as any account, but I think this could be achieved in a number of ways, not sure what's best for now, so I just reworded the title of the issue. In particular, the ongoing assumption was that the calling account is dependent on the connection - another option would be to have a method like `as_dry_run(account_id)` on the connection itself that would return a `ReadOnlyConnection` with the account to use baked in. TBD by person working on this issue.
